pkg/stdout/capture.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package stdout
+
+import (
+	"bufio"
+	"context"
+	"log"
+	"os"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+func Capture(ctx context.Context, inner func() error) error {
+	r, _, cleanup := mustHijackStdOut()
+	scanner := bufio.NewScanner(r)
+	go func() {
+		for scanner.Scan() {
+			if ctx.Err() != nil {
+				return
+			}
+			logEntry := scanner.Text()
+			parsedLogEntry := parseLogEntry(logEntry)
+			if parsedLogEntry.IsRawString {
+				_, err := logrus.StandardLogger().Out.Write([]byte(parsedLogEntry.Message + "\n"))
+				if err != nil {
+					return
+				}
+				continue
+			}
+			logrus.WithField("dcgm_level", parsedLogEntry.Level).Info(parsedLogEntry.Message)
+		}
+		if err := scanner.Err(); err != nil {
+			log.Printf("Error reading from pipe: %v", err)
+		}
+	}()
+
+	defer func() {
+		cleanup()
+	}()
+
+	// Call function here
+	return inner()
+}
+
+func mustHijackStdOut() (r *os.File, o *os.File, cleanup func()) {
```

Comment:
Please add explanation of the func `mustHijackStdOut` in Comment above.

Naming preference: `captureStdOut` instead of `mustHijackStdOut`

Reply:
done